src/semaphore.c

```diff
@@ -716,6 +771,11 @@ _dispatch_thread_semaphore_create(void)
 	int ret = sem_init(s4, 0, 0);
 	DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 	return (_dispatch_thread_semaphore_t)s4;
+#elif USE_FUTEX_SEM
+	dispatch_futex_t *futex = malloc(sizeof(*futex));
```

Comment:
actually unlike other stuff that needs to be in the same struct for performance and also to be able to run a destructor, we don't care for a futex on linux.

given that `__thread` usage is fine on linux, just have a global:

static __thread struct _dispatch_futex_s _dispatch_thread_semaphore;

and avoid this malloc altogether.